jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2StreamEndPoint.java

```diff
@@ -342,6 +344,22 @@ public void write(Callback callback, ByteBuffer... buffers) throws WritePendingE
         }
     }
 
+    @Override
+    public Callback cancelWrite(Throwable cause)
+    {
+        try (Blocker.Callback blocker = Blocker.callback())
+        {
+            stream.reset(new ResetFrame(stream.getId(), ErrorCode.CANCEL_STREAM_ERROR.code), blocker);
+            blocker.block();
```

Comment:
I don't think this is right.

The method `write()` above this method initiates a write by calling `stream.data()`; we should "steal" the application callback like we do in `WriteFlusher`.

Reply:
@sbordet I don't think we can steal the CB pass to stream.data, but we could steal the CB wrapped inside of the `dataCallback` above.  Is that what you are thinking?

Reply:
@gregw yes, because the wrapped CB would be the one that eventually releases the buffer, and we need to ensure that it is completed after the reset.